scene/main/viewport.cpp

```diff
@@ -993,11 +994,6 @@ void Viewport::set_world_2d(const Ref<World2D> &p_world_2d) {
 		return;
 	}
 
-	if (parent && parent->find_world_2d() == p_world_2d) {
-		WARN_PRINT("Unable to use parent world_2d as world_2d");
-		return;
-	}
-
```

Comment:
Was this removed intentionally?

Reply:
yes, it fixes this https://github.com/godotengine/godot/issues/31856, since you can use layers and masks to prevent any issues that were previously prevented by this check

Reply:
probably worth a comment explaining this then

Reply:
Sorry, where should I comment?

Reply:
Commenting in the PR description is fine